libipl/p10/common.C

```diff
@@ -232,3 +231,59 @@ void ipl_log_sbe_ffdc(struct pdbg_target *pib)
 	if (ffdc)
 		free(ffdc);
 }
+
+int ipl_set_sbe_state(struct pdbg_target *proc, enum sbe_state state)
+{
+	ipl_log(IPL_INFO, "(%s)SBE state update (%d) \n",
+		pdbg_target_path(proc), state);
+
+	// get PIB target
+	char path[16];
+	sprintf(path, "/proc%d/pib", pdbg_target_index(proc));
+	struct pdbg_target *pib = pdbg_target_from_path(nullptr, path);
+	if (pib == nullptr) {
+		ipl_log(IPL_ERROR, "Failed to get PIB target for(%s)",
+			pdbg_target_path(proc));
+		ipl_error_callback(IPL_ERR_PIB_TGT_NOT_FOUND);
+		return 1;
+	}
+	//PIB already probed as part of ipl init function.
+	//update SBE state
+	if (sbe_set_state(pib, state)) {
+		ipl_log(IPL_ERROR, "Failed to update SBE state information (%s)",
+			pdbg_target_path(proc));
+		ipl_error_callback(IPL_ERR_CFAM);
```

Comment:
is this correct error? should we add a new error

Reply:
SBE State function failure mainly due to fsi read/write failure . For future error callout handling used generic word "CFAM" . will change to IPL_ERR_FSI.

Reply:
k